lib/fields.py

```diff
@@ -109,26 +119,28 @@ class UsersSearchSelectField(SearchSelectField):
     to this field must be of type UserProfile, not User, and likewise,
     the corresponding model field must be a ManyToManyField to UserProfile.
     """
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
         # to_field_name: use User IDs in the HTML widget since the queryset
         # is attached to UserProfiles.
         # to_field_name affects both the value attributes in the HTML form and
         # the validation of the user's input.
         kwargs['to_field_name'] = 'user_id' # userprofile.user.id
+        if not 'widget' in kwargs:
+            # Create a search select widget specialized for users.
+            kwargs['widget'] = SearchSelect(
+                ajax=True,
+                display_fields=['full_name', 'student_id', 'email'],
+                clipboard_fields=['student_id', 'email'],
+                field_sources={
+                    'full_name': 'user.get_full_name',
+                    'email': 'user.email',
+                },
+                field_labels={
+                    'student_id': 'Student IDs',
+                    'email': 'Email addresses',
+                },
+            )
```

Comment:
As seen here, the `SearchSelect` widget takes a bunch of lists and dicts as parameters when you want to customize it. The other option was to take one list of instances of a custom class, which defines the necessary attributes for a single field (something like `name, source, label, is_displayed, is_copyable`). I ended up not going with that option, but this is something I debated for a long time.